### PR TITLE
Add ZKsync Era network support.

### LINKED ISSUE #1371: ZKsync network service support
The Infura service is adding support for ZKsync Era via DIN.

### PR DESCRIPTION
**Pull request by bgravenorst**
# Description

ZKsync Era is now supported by the Infura service.

## Issue(s) fixed

#1371

## Checklist

Complete this checklist before merging your PR:

- [x] If this PR contains a major change to the documentation content, I have added an entry to the top of the ["What's new?"](https://github.com/MetaMask/metamask-docs/blob/main/docs/whats-new.md) page.
- [x] The proposed changes have been reviewed and approved by a member of the documentation team.

### CODE REVIEW

**Comment by joaniefromtheblock** [comment]
Added changes/suggestions from this [PR](https://github.com/INFURA/docs/pull/248)

**Comment by joaniefromtheblock** [comment]
Need to add update [specified in chat ](https://consensys.slack.com/archives/C06HPSUP1S6/p1719844236292519?thread_ts=1719842761.494629&cid=C06HPSUP1S6) and include another mention to Services docs in the What's New section

File: services/get-started/endpoints.md [hunk 1]
```diff
@@ -143,6 +143,12 @@ Sepolia instead.
 | Mainnet           | JSON-RPC over HTTPS | `https://starknet-mainnet.infura.io/v3/<API-KEY>` |
 | Testnet (Sepolia) | JSON-RPC over HTTPS | `https://starknet-sepolia.infura.io/v3/<API-KEY>` |
 
+## ZkSync Era
```
Comment (alexandratran) [inline services/get-started/endpoints.md:146]:
```suggestion
## ZKsync Era
```

File: services/reference/blast/quickstart.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ In these examples, you'll use [NPM](https://docs.npmjs.com/downloading-and-insta
 
 #### Node Fetch
 
-1. In your project folder, install the Node Fetch package using `npm`:
+1. In your project folder, install the `node-fetch package using `npm`:
```
Comment (alexandratran) [inline services/reference/blast/quickstart.md:39]:
```suggestion
1. In your project folder, install the `node-fetch` package using `npm`:
```

File: services/reference/zksync/json-rpc-methods/_eth_accounts-request.mdx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+```bash
+curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+    -d '{"jsonrpc":"2.0","method":"eth_accounts","params":[],"id":1}'
+```
+
+  </TabItem>
+
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/_eth_accounts-request.mdx:15]:
````suggestion
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc":"2.0","method":"eth_accounts","params":[],"id":1}'
```

  </TabItem>
````

File: services/reference/zksync/json-rpc-methods/_eth_blocknumber-request.mdx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+```bash
+curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+    -d '{"jsonrpc":"2.0","method":"eth_blockNumber","params": [],"id":1}'
+```
+
+  </TabItem>
+
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/_eth_blocknumber-request.mdx:15]:
````suggestion
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc":"2.0","method":"eth_blockNumber","params": [],"id":1}'
```

  </TabItem>
````

File: services/reference/zksync/json-rpc-methods/_eth_call-request.mdx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+  <TabItem value="cURL">
+
+  ```bash
+  curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+    -d '{"jsonrpc":"2.0","method":"eth_call","params": [{"from": "0xb60e8dd61c5d32be8058bb8eb970870f07233155","to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567","gas": "0x76c0","gasPrice": "0x9184e72a000","value": "0x9184e72a","data": "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675"}, "latest"],"id":1}'
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/_eth_call-request.mdx:11]:
````suggestion
```bash
curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc":"2.0","method":"eth_call","params": [{"from": "0xb60e8dd61c5d32be8058bb8eb970870f07233155","to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567","gas": "0x76c0","gasPrice": "0x9184e72a000","value": "0x9184e72a","data": "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675"}, "latest"],"id":1}'
````

File: services/reference/zksync/json-rpc-methods/_eth_chainid-request.mdx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+  <TabItem value="cURL">
+
+```bash
+curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+    -d '{"jsonrpc":"2.0","method":"eth_chainId","params": [],"id":1}'
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/_eth_chainid-request.mdx:11]:
```suggestion
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc":"2.0","method":"eth_chainId","params": [],"id":1}'
```

File: services/reference/zksync/json-rpc-methods/_eth_estimategas-request.mdx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+<Tabs>
+  <TabItem value="cURL">
+
+  ```bash
+  curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+          -X POST \
+          -H "Content-Type: application/json" \
+          -d '{"jsonrpc":"2.0","method":"eth_estimateGas","params": [{"from": "0x9cE564c7d09f88E7d8233Cdd3A4d7AC42aBFf3aC","to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567","value": "0x9184e72a"}],"id":1}'
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/_eth_estimategas-request.mdx:12]:
````suggestion
```bash
curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc":"2.0","method":"eth_estimateGas","params": [{"from": "0x9cE564c7d09f88E7d8233Cdd3A4d7AC42aBFf3aC","to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567","value": "0x9184e72a"}],"id":1}'
```
````

File: services/reference/zksync/json-rpc-methods/_eth_getblockbyhash-request.mdx [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+<Tabs>
+  <TabItem value="cURL">
+
+  ```bash
+  curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+      -X POST \
+      -H "Content-Type: application/json" \
+      -d '{"jsonrpc":"2.0","method":"eth_getBlockByHash","params": ["0xb3b20624f8f0f86eb50dd04688409e5cea4bd02d700bf6e79e9384d47d6a5a35",false],"id":1}'
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/_eth_getblockbyhash-request.mdx:12]:
````suggestion
```bash
curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc":"2.0","method":"eth_getBlockByHash","params": ["0xb3b20624f8f0f86eb50dd04688409e5cea4bd02d700bf6e79e9384d47d6a5a35",false],"id":1}'
```
````

File: services/reference/zksync/json-rpc-methods/zks_estimatefee.mdx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+<Tabs>
+  <TabItem value="cURL">
+
+  ```bash
+   curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+    -d '{"jsonrpc":"2.0", "id":2, "method": "zks_estimateFee", "params": [ { "from": "0x1111111111111111111111111111111111111111", "to":"0x2222222222222222222222222222222222222222", "data": "0xffffffff" } ] }'
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/zks_estimatefee.mdx:39]:
````suggestion
```bash
curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc":"2.0", "id":2, "method": "zks_estimateFee", "params": [ { "from": "0x1111111111111111111111111111111111111111", "to":"0x2222222222222222222222222222222222222222", "data": "0xffffffff" } ] }'
```
````

File: services/reference/zksync/json-rpc-methods/zks_estimatefee.mdx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    "jsonrpc": "2.0",
+    "result": {
+        "gas_limit": "0x156c00",
+        "gas_per_pubdata_limit": "0x143b",
+        "max_fee_per_gas": "0xee6b280",
+        "max_priority_fee_per_gas": "0x0"
+    },
+    "id": 2
+  }
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/zks_estimatefee.mdx:58]:
````suggestion

```json
{
  "jsonrpc": "2.0",
  "result": {
    "gas_limit": "0x156c00",
    "gas_per_pubdata_limit": "0x143b",
    "max_fee_per_gas": "0xee6b280",
    "max_priority_fee_per_gas": "0x0"
  },
  "id": 2
}
```

````

File: services/reference/zksync/json-rpc-methods/zks_getallaccountbalances.mdx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+<Tabs>
+  <TabItem value="cURL">
+
+  ```bash
+    curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+    -d '{"jsonrpc": "2.0", "id": 1, "method": "zks_getAllAccountBalances", "params": [ "0x98E9D288743839e96A8005a6B51C770Bbf7788C0" ]}'
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/zks_getallaccountbalances.mdx:32]:
````suggestion
```bash
curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc": "2.0", "id": 1, "method": "zks_getAllAccountBalances", "params": [ "0x98E9D288743839e96A8005a6B51C770Bbf7788C0" ]}'
```

````

File: services/reference/zksync/json-rpc-methods/zks_getallaccountbalances.mdx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+  <TabItem value="JSON">
+  ```json
+  {
+    "jsonrpc": "2.0",
+    "result": {
+        "0x0000000000000000000000000000000000000000": "0x2fbd72a1121b3100"
+    },
+    "id": 2
+  }
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/zks_getallaccountbalances.mdx:48]:
````suggestion

```json
{
  "jsonrpc": "2.0",
  "result": {
    "0x0000000000000000000000000000000000000000": "0x2fbd72a1121b3100"
  },
  "id": 2
}
```

````

File: services/reference/zksync/json-rpc-methods/zks_getbridgecontracts.mdx [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+<Tabs>
+  <TabItem value="cURL">
+
+  ```bash
+    curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+    -d '{"jsonrpc": "2.0", "id": 1, "method": "zks_getBridgeContracts", "params": []}' \
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/zks_getbridgecontracts.mdx:37]:
````suggestion
```bash
curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc": "2.0", "id": 1, "method": "zks_getBridgeContracts", "params": []}' \
```

````

File: services/reference/zksync/json-rpc-methods/zks_getbytecodebyhash.mdx [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+        0,
+        2,
+        0,
+        11,
+        ...,
+        ...
+    ],
+    "id": 1
+  }
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/zks_getbytecodebyhash.mdx:60]:
````suggestion

```json
{
  "jsonrpc": "2.0",
  "result": [
    0,
    4,
    0,
    0,
    0,
    0,
    0,
    2,
    0,
    11,
    ...
  ],
  "id": 1
}
```

````

File: services/reference/zksync/json-rpc-methods/zks_getrawblocktransactions.mdx [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+
+<Tabs>
+  <TabItem value="cURL">
+
+  ```bash
+    curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+   -d '{"jsonrpc": "2.0", "id": 1, "method": "zks_getRawBlockTransactions", "params": [5817]}'
+  ```
```
Comment (alexandratran) [inline services/reference/zksync/json-rpc-methods/zks_getrawblocktransactions.mdx:55]:
````suggestion
```bash
curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc": "2.0", "id": 1, "method": "zks_getRawBlockTransactions", "params": [5817]}'
```

````

File: services/reference/zksync/quickstart.md [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+
+### cURL
+
+Run the following command in your terminal, replacing `YOUR-API-KEY` with your actual Infura API key:
+
+```bash
+curl https://zksync-mainnet.infura.io/v3/YOUR-API-KEY \
+    -X POST \
+    -H "Content-Type: application/json" \
+    -d '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}'
```
Comment (alexandratran) [inline services/reference/zksync/quickstart.md:30]:
```suggestion
  -X POST \
  -H "Content-Type: application/json" \
  -d '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}'
```

File: services/reference/zksync/quickstart.md [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+1. Create your JavaScript file and copy the following code:
+
+    Replace `YOUR-API-KEY` with your actual Infura API key.
+
+    ```javascript title="index.js"
+    const axios = require('axios');
+    
+    axios.post('https://zksync-mainnet.infura.io/v3/YOUR-API-KEY', {
+      jsonrpc: '2.0',
+      method: 'eth_blockNumber',
```
Comment (alexandratran) [inline services/reference/zksync/quickstart.md:98]:
```suggestion
    const axios = require("axios");
    
    axios.post("https://zksync-mainnet.infura.io/v3/YOUR-API-KEY", {
      jsonrpc: "2.0",
      method: "eth_blockNumber",
```

File: services/reference/zksync/quickstart.md [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+    ``````
+
+1. Create your JavaScript file and copy the following code:
+
+    Replace `YOUR-API-KEY` with your actual Infura API key.
+
+    ```javascript title="index.js"
+    const ethers = require('ethers');
+
+    const provider = new ethers.providers.JsonRpcProvider('https://zksync-mainnet.infura.io/v3/YOUR-API-KEY');
```
Comment (alexandratran) [inline services/reference/zksync/quickstart.md:131]:
```suggestion
    const ethers = require("ethers");

    const provider = new ethers.providers.JsonRpcProvider("https://zksync-mainnet.infura.io/v3/YOUR-API-KEY");
```

File: services/reference/zksync/quickstart.md [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+    url = 'https://zksync-mainnet.infura.io/v3/YOUR-API-KEY'
+
+    payload = {
+        "jsonrpc": "2.0",
+        "method": "eth_blockNumber",
+        "params": [],
+        "id": 1
+    }
+
+    headers = {'content-type': 'application/json'}
```
Comment (alexandratran) [inline services/reference/zksync/quickstart.md:173]:
```suggestion
    url = "https://zksync-mainnet.infura.io/v3/YOUR-API-KEY"

    payload = {
      "jsonrpc": "2.0",
      "method": "eth_blockNumber",
      "params": [],
      "id": 1
    }

    headers = {"content-type": "application/json"}
```

**Review by alexandratran — COMMENTED** [review]
Suggestions to address some code sample style issues. Also raised [this issue](https://github.com/MetaMask/metamask-docs/issues/1375) to do a more thorough audit in a future PR.

**Review by joaniefromtheblock — APPROVED** [review]
LGTM

**Pull request merged**